.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,288 @@
+          rustup toolchain install stable
+          rustup toolchain install nightly
+          rustup default stable
+          rustup target add wasm32-unknown-unknown --toolchain nightly
+          cargo +nightly install --git https://github.com/alexcrichton/wasm-gc --force
+          rustup update
+      -
+        name: build
+        env:
+          RUST_BACKTRACE: full
+          RUSTC_WRAPPER: sccache
+          SCCACHE_CACHE_SIZE: 2G
+          SCCACHE_DIR: /home/runner/.cache/sccache
+          CARGO_TERM_COLOR: always
+        run: |
+          source ${HOME}/.cargo/env
+          cargo install --verbose --git https://github.com/${GITHUB_REPOSITORY} --rev ${GITHUB_SHA:0:7} --features runtime-benchmarks --locked --force --root ~/.cargo
+      -
+        name: stop sccache server
+        run: sccache --stop-server || true
+      -
+        name: upload
+        uses: actions/upload-artifact@v2
+        with:
+          name: manta
+          path: /home/runner/.cargo/bin/manta
+
+  run-benchmark:
+    name: benchmark (${{ matrix.benchmark.pallet.name }} ${{ matrix.benchmark.extrinsic.name }} < ${{ matrix.benchmark.expected.time }} ${{ matrix.benchmark.expected.unit }})
+    needs:
+      - start-worker
+      - build-benchmark
+    runs-on: ${{ needs.start-worker.outputs.runner-label }}
+    strategy:
+      fail-fast: false
+      matrix:
+        benchmark:
+          -
+            extrinsic:
+              id: init_asset
+              name: init-asset
+            pallet:
+              id: pallet_manta_pay
+              name: pallet-manta-pay
+            iterations: 1000
+            expected:
+              time: 6000
+              unit: µs
+          -
+            extrinsic:
+              id: transfer_asset
+              name: transfer-asset
+            pallet:
+              id: pallet_manta_pay
+              name: pallet-manta-pay
+            iterations: 1000
+            expected:
+              time: 60
+              unit: µs
+          -
+            extrinsic:
+              id: mint_private_asset
+              name: mint-private-asset
+            pallet:
+              id: pallet_manta_pay
+              name: pallet-manta-pay
+            iterations: 1000
+            expected:
+              time: 48000
+              unit: µs
+          -
+            extrinsic:
+              id: private_transfer
+              name: private-transfer
+            pallet:
+              id: pallet_manta_pay
+              name: pallet-manta-pay
+            iterations: 1000
+            expected:
+              time: 220000
+              unit: µs
+          -
+            extrinsic:
+              id: reclaim
+              name: reclaim
+            pallet:
+              id: pallet_manta_pay
+              name: pallet-manta-pay
+            iterations: 1000
+            expected:
+              time: 180000
+              unit: µs
+    steps:
+      -
+        name: install mongodb-mongosh
+        run: |
```

Comment:
mind elaborating why we need mongodb here?

Reply:
not at all. each time we run benchmarks we [push a benchmark stats record](https://github.com/Manta-Network/Manta/blob/44a2ff9/.github/workflows/benchmark.yml#L204-L224) to a mongo db. this enables the historical benchmark stats to be available for analysis over time or across changes. eg: https://dev.stats.manta.systems